controllers/statestorage_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/alexedwards/scs/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVeilSPIState(t *testing.T) {
+	//given
+	req := httptest.NewRequest("GET", fmt.Sprintf("/?state=%s&k8s_token=%s", "statestr", "token-234234"), nil)
+	res := httptest.NewRecorder()
+	sessionManager := scs.New()
+	storage := NewStateStorage(sessionManager)
+
+	//when
+	sessionManager.LoadAndSave(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		newStateString, err := storage.VeilRealState(r)
+		assert.NoError(t, err)
+		assert.True(t, "statestr" != newStateString)
+		assert.Equal(t, 32, len(newStateString))
+		assert.Equal(t, "statestr", sessionManager.Get(r.Context(), newStateString))
+
+	})).ServeHTTP(res, req)
+
+	//then
+	assert.Equal(t, 1, len(res.Result().Cookies()))
+	assert.Equal(t, "session", res.Result().Cookies()[0].Name)
+	assert.NotEmpty(t, res.Result().Cookies()[0].Value)
+
+}
+func Test_FailToVeilIfStateIsEmpty(t *testing.T) {
+	//given
+	req := httptest.NewRequest("GET", fmt.Sprintf("/?state=%s&k8s_token=%s", "", "token-234234"), nil)
+	res := httptest.NewRecorder()
+	sessionManager := scs.New()
+	storage := StateStorage{sessionManager}
+
+	//when
+	sessionManager.LoadAndSave(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		newStateString, err := storage.VeilRealState(r)
+		if assert.Error(t, err) {
+			assert.Equal(t, errors.New("request has no `state` parameter"), err)
```

Comment:
Something like
```
assert.True(t, errors.Is(err, noStateError))
```
would IMHO read better and would also survive wrapping this error if some future code change did that.